lxd/metrics/metrics.go

```diff
@@ -24,26 +22,19 @@ func NewMetricSet(labels map[string]string) *MetricSet {
 	return &out
 }
 
-// FilterSamples filters the existing MetricSet using the given permission checker. Samples not containing the "project" label are skipped.
-func (m *MetricSet) FilterSamples(permissionChecker func(entityURL *api.URL) bool) {
+// FilterSamples filters the existing MetricSet using the given permission checker.
+func (m *MetricSet) FilterSamples(metricFilters []MetricFilter) {
 	for metricType, samples := range m.set {
 		allowedSamples := make([]Sample, 0, len(samples))
 		for _, s := range samples {
-			projectName := s.Labels["project"]
-			instanceName := s.Labels["name"]
-			if projectName == "" {
-				continue
-			}
-
-			var hasPermission bool
-
-			if instanceName != "" {
-				hasPermission = permissionChecker(entity.InstanceURL(projectName, instanceName))
-			} else {
-				hasPermission = permissionChecker(entity.ProjectURL(projectName))
+			permittedSample := false
+			for _, filter := range metricFilters {
+				if filter.PermissionChecker(filter.EntityURL(s.Labels)) {
```

Comment:
Did you consider having `FilterSamples()` accept a `permissionChecker func(labels map[string]string) bool` type function, which would combine the call to `filter.PermissionChecker(filter.EntityURL(s.Labels)) ` internally rather than having to add `type MetricFilter struct`?

Effectively should we have a function that takes a set of labels and converts them to a URL suitable for use with the `PermissionChecker`?

Reply:
Ah thanks! Like it, let me add a fix.

Reply:
Fixed, looks much better now :)